src/main.c

```diff
@@ -114,16 +114,34 @@ activate Sonic mode.\n");
     return 0;
 }
 
+static char *get_XDG_CONFIG_HOME()
+{
+    const char *default_config_subdir = "/.config";
+    char *config_dir;
+    char *xdg = getenv("XDG_CONFIG_HOME");
+    if (!xdg) {
+        char *user_home = getenv("HOME");
+        if (!user_home) {
+            lprintf(warning, "$HOME is unset\n");
+            char *cur_dir = realpath("./", NULL);
+            if (!cur_dir) {
+                lprintf(fatal, "Could not get config directory\n");
```

Comment:
This should not be fatal, if the software cannot get the config directory, it should keep running. This should just be a warning.